### DISCUSSION TITLE
Meeting 2024-07-02

### DISCUSSION THREAD

**Discussion by adamgreig** [rust-embedded/wg#772] [General]
This is the agenda for today's meeting. Please add anything you'd like to discuss below!

* Meetings: Tuesday 8pm Europe/Berlin time
* [Join the Matrix chat](https://matrix.to/#/#rust-embedded:matrix.org)
* [Nominated issues](https://github.com/search?q=org%3Arust-embedded+label%3Anominated+is%3Aopen&type=Issues)
* [IRC/Matrix logs](https://libera.irclog.whitequark.org/rust-embedded/2024-07-02)

### Announcements

* Council meeting this week so please let @jamesmunns know if you have any items for them. No other update on the proposed new Rust teams organisation from last meeting.

**Comment by adamgreig** [comment]
Discovery book rewrite RFC: https://github.com/rust-embedded/wg/pull/770

**Reply by adamgreig** [reply]
We discussed this and everyone present was in favour. We discussed briefly what to do about translations without any firm decisions at the moment, noting that currently we just link to externally forked translations, but there has been discussion about using gettext-style translation tooling to make it easier to make and host translations.

It was clarified that the existing URLs will all remain unchanged with the existing material still hosted, but clearly marked as deprecated and linking to the new URL with the new book.

**Comment by adamgreig** [comment]
Async CAN API: https://github.com/rust-embedded/embedded-hal/pull/585

**Reply by adamgreig** [reply]
* @Dirbaio commented to point out a change needed for the `cfg` attributes
* Otherwise general approval
* We discussed having some sort of read_ready/write_ready method to allow "non-blocking" async use, akin to the earlier try_ methods, and so far the conclusion is still to leave them out

**Comment by adamgreig** [comment]
From @jamesmunns for the resources team:

We've received a PR for adding commercially available books ("Simplified Embedded Rust" books from apollo labs): https://github.com/rust-embedded/awesome-embedded-rust/pull/453

 * To date, we've not had any "paid only" material on the list
 * We HAVE had "paid adjacent" material, e.g. training material from Ferrous, TweedeGolf, etc., where the MATERIAL was public, but was for commercial purposes
 * IMO the choices are:
    * We codify that we don't accept private commercial items
    * We set up a separate section for paid material, and merge this PR in that section

**Reply by adamgreig** [reply]
We didn't get time to discuss this point this week.

**Reply by jamesmunns** [reply]
CC @rust-embedded/resources team, does anyone have objections to "make a separate section for paid stuff"?

**Reply by adamgreig** [reply]
I'm happy to have a section for paid material. If we end up unbalanced or with too much stuff we don't want we can always revisit it.

**Reply by eldruin** [reply]
I am also fine with a separate section as long as we stay balanced as stated above.

**Reply by jamesmunns** [reply]
Okay, sounds like we have at least 3/6 folks chiming in, I'll tentatively move ahead with this, if anyone objects we can revert the PR.

**Reply by jamesmunns** [reply]
Opened https://github.com/rust-embedded/awesome-embedded-rust/pull/455

**Comment by adamgreig** [comment]
From @jamesmunns for the cortex-m team:

* Poke the cortex-m team to see if it's reasonable to add a way to eagerly detect when we are missing a "real" linker script, e.g. force a linker error for a missing symbol defined in the c-m-rt link.x and referenced by the c-m-rt crate, so people don't end up with "no loadable section" errors from probe-rs.

* I feel like I've seen people asking about that more recently, not sure if anything has changed, but it feels like something we can do something about, at least in the typical case.

**Reply by adamgreig** [reply]
We didn't get time to discuss this point this week.

**Comment by romancardenas** [comment]
RISC-V fallible functions are almost done: https://github.com/rust-embedded/riscv/pull/222

**Reply by adamgreig** [reply]
We didn't get time to discuss this point this week.
